amqp/src/test/scala/akka/stream/alpakka/amqp/scaladsl/AmqpConnectionProvidersSpec.scala

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.amqp.scaladsl
+
+import akka.stream.alpakka.amqp._
+
+class AmqpConnectionProvidersSpec extends AmqpSpec {
+  "The AMQP Default Connection Provider" should {
+    "create a new connection per invocation of DefaultAmqpConnection" in {
+      val connectionProvider = DefaultAmqpConnectionProvider.create(DefaultAmqpConnection)
+      val connection1 = connectionProvider.get
+      val connection2 = connectionProvider.get
+      connection1 should not equal connection2
+      connectionProvider.release(connection1)
+      connectionProvider.release(connection2)
```

Comment:
As these calls to get and release can happen concurrently, I think it would be valuable if the tests were concurrent as well. Will not be deterministic but would hopefully have caught the concurrency bug I noted above.